frame/clients/drag_and_drop_demo.py

```diff
@@ -0,0 +1,216 @@
+import gi
+import sys
+
+gi.require_version("Gtk", "3.0")
+from gi.repository import Gtk, Gdk, GdkPixbuf
+
+(EXCHANGE_TYPE_NONE, EXCHANGE_TYPE_TEXT, EXCHANGE_TYPE_PIXBUF) = range(3)
+(TARGET_ENTRY_TEXT, TARGET_ENTRY_PIXBUF) = range(2)
+(COLUMN_TEXT, COLUMN_PIXBUF) = range(2)
+
+DRAG_ACTION = Gdk.DragAction.COPY
+
+
+class DragDropWindow(Gtk.Window):
+    def __init__(self, source_mode, target_mode, expect):
```

Comment:
An easier read, IMO:
```suggestion
class DragDropWindow(Gtk.Window):
    expect = EXCHANGE_TYPE_NONE
    result = EXCHANGE_TYPE_NONE
    
    def __init__(self, source_mode, target_mode, expect=EXCHANGE_TYPE_NONE):
        self.expect = expect
```